src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,66 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { closeCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
```

Comment:
The `getUser` function should handle errors in case the API call fails. Consider adding a `.catch` block to manage potential errors and provide feedback to the user.